### PR TITLE
Refactor wrapper for atomic crypto transfer

### LINKED ISSUE #4085: Refactor wrapper classes that contain the parameters for the cryptoTransfer call
### Problem

Epic: #4084 

The current wrapper classes only support the transfer of fungible and non-fungible tokens.  We need to add support for hbar transfers as well. 

### Solution

Create a new wrapper that wraps both the existing token transfer wrapper while adding a new wrapper for hbar transfers.  Update existing decoders to return the new wrapper class.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
The wrapper class for cryptoTransfer needs to be refactored in order to support hbar transfers.

Epic: #4084 

**Related issue(s)**: #3758 

Fixes #4085 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC721PrecompilesTest.java [hunk 1]
```diff
@@ -1720,6 +1722,11 @@ private Bytes givenMinimalFrameContextWithoutParentUpdater(Bytes nestedPretendAr
                     List.of(new SyntheticTxnFactory.NftExchange(1, token, sender, receiver)),
                     new ArrayList<>() {});
 
+    public static final CryptoTransferWrapper CRYPTO_TRANSFER_TOKEN_WRAPPER =
+            new CryptoTransferWrapper(
+                    new TransferWrapper(Collections.emptyList()),
+                    Collections.singletonList(TOKEN_TRANSFER_WRAPPER));
+
```
Comment (dimitar-dinev) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC721PrecompilesTest.java:1729]: Same question as in `ERC20PrecompilesTest`:

Can't we directly use the new constants defined in `HTSTestsUtil`, removing the need for the new ones here?

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC20PrecompilesTest.java [hunk 2]
```diff
@@ -1701,6 +1703,10 @@ private void givenPricingUtilsContext() {
                             new SyntheticTxnFactory.FungibleTokenTransfer(
                                     -AMOUNT, false, token, sender, null)));
 
+    public static final CryptoTransferWrapper CRYPTO_TRANSFER_TOKEN_WRAPPER =
+            new CryptoTransferWrapper(
+                    new TransferWrapper(Collections.emptyList()),
+                    Collections.singletonList(TOKEN_TRANSFER_WRAPPER));
```
Comment (dimitar-dinev) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC20PrecompilesTest.java:1709]: Can't we directly use the new constants defined in `HTSTestsUtil`, removing the need for the new ones here?

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java [hunk 3]
```diff
@@ -221,18 +226,22 @@ public static List<TokenTransferWrapper> decodeERCTransfer(
         addSignedAdjustment(fungibleTransfers, token, recipient, amount.longValueExact());
         addSignedAdjustment(fungibleTransfers, token, caller, -amount.longValueExact());
 
-        return Collections.singletonList(
-                new TokenTransferWrapper(NO_NFT_EXCHANGES, fungibleTransfers));
+        final List<TokenTransferWrapper> tokenTransferWrappers =
```
Comment (dimitar-dinev) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java:229]: nit: can use `var` here and in the rest repeated places

**Review by dimitar-dinev — APPROVED** [review]
LGTM! Just a few non-blocking nits

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/codec/CryptoTransferWrapper.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2020-2022 Hedera Hashgraph, LLC
```
Comment (Nana-EC) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/codec/CryptoTransferWrapper.java:2]:
```suggestion
 * Copyright (C) 2022 Hedera Hashgraph, LLC
```

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC20PrecompilesTest.java (hunk 2 above)
Comment (lukelee-sl) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC20PrecompilesTest.java]: done

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC721PrecompilesTest.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/ERC721PrecompilesTest.java]: done

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java (hunk 3 above)
Comment (lukelee-sl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCTransferPrecompile.java]: done

**Review by shemnon — APPROVED** [review]
LGTM

**Pull request merged**